src/tools/rbd_mirror/image_sync/ImageCopyRequest.cc

```diff
@@ -146,7 +148,20 @@ void ImageCopyRequest<I>::send_object_copies() {
       }
     }
     complete = (m_current_ops == 0);
+
+    if (!complete) {
+      m_update_sync_ctx = new FunctionContext([this](int r) {
+          this->send_update_sync_point();
+      });
+    }
+  }
+
+  if (m_update_sync_ctx) {
+    Mutex::Locker l(*m_timer_lock);
+    m_timer->add_event_after(m_update_sync_point_interval,
```

Comment:
Yep, that's what I said in my comment above, and already fixed in the implementation, although still didn't pushed the changes